cirq/protocols/resolve_parameters.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Any, TypeVar
+from typing_extensions import Protocol
+from cirq.study import ParamResolver
+
+TDefault = TypeVar('TDefault')
+
+class SupportsParameterization(Protocol):
+    """An object that can be parameterized by Symbols and resolved
+    via a ParamResolver"""
+
+    def _is_parameterized_(self: Any) -> bool:
+        """Whether the gate is parameterized by any Symbols that require
+        resolution.  Returns True if the gate has any unresolved Symbols
+        and False otherwise."""
+
+    def _resolve_parameters_(self: Any, param_resolver: ParamResolver):
+        """Resolve the parameters in the effect."""
+
+
+
+def is_parameterized(val: Any) -> bool:
+    """Returns whether the object is parameterized with any Symbols.
+
+    A value is parameterized when it has an is_parameterized method and
```

Comment:
```
an `_is_parameterized_` method
```

The underscores are extremely important to include when referring to the method.